tests/unit/mathematics/linalg/Memory_transfer_unittest.cc

```diff
@@ -0,0 +1,80 @@
+#include <shogun/lib/config.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+#ifdef HAVE_VIENNACL
+#include <shogun/mathematics/linalg/LinalgBackendViennaCL.h>
+#endif
+
+#include <shogun/lib/SGVector.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+using namespace linalg;
+
+TEST(LinalgBackendViennaCL, SGVector_to_gpu_viennacl)
+{
+#ifdef HAVE_VIENNACL
+	sg_linalg->set_gpu_backend(new LinalgBackendViennaCL());
+#endif
+
+	const index_t size = 10;
+	SGVector<int32_t> a(size), b(size);
+	a.range_fill(0);
+
+	b = to_gpu(a);
+
+	if (sg_linalg->get_gpu_backend())
+	{
+		EXPECT_EQ(a.on_gpu(), false);
+		EXPECT_EQ(b.on_gpu(), true);
+	}
+	else
+	{
+		EXPECT_EQ(a.on_gpu(), false);
+		EXPECT_EQ(b.on_gpu(), false);
+	}
+}
```

Comment:
@karlnapf If I group the examples by backend, I am wondering where this examples should go? Actually it is only testing for linalg namespace method on Travis now but not BackendEigen or BackendViennaCL.

Can I 
- Leave the memory transfer example intact (or this is what you meant by "without backend specific operations"? )
- Group other operations by backend
  - Eigen3 
  - ViennaCL with all tests guarded by `#ifdef HAVE_VIENNACL`